js/libs/form.js

```diff
@@ -1495,6 +1515,11 @@ Fliplet().then(function() {
                       });
 
                       field.value = options;
+                    } else if (field._type === 'flTypeahead') {
+                      getDataSourceColumnValues(field).then(function(result) {
```

Comment:
```js
                      getDataSourceColumnValues(field).then(function(result) {
                        field.options = result;
                        field.value = value;
                      });
```

You've used the pattern above several times. I suggest creating a `setTypeaheadFieldValue(field, value)` function to do this